compiler-core/src/parse/lexer.rs

```diff
@@ -617,6 +617,18 @@ where
         if self.chr0 == Some('.') {
             value.push(self.next_char().expect("lex_normal_number float"));
             value.push_str(&self.radix_run(10));
+
+            // If scientific:
```

Comment:
I wasn't sure if `lex_normal_number` should be renamed, or if the scientific notation stuff should be pulled out into a separate function.

Reply:
I think `lex_decimal_number` might be a more appropriate name?